### PR TITLE
[WIP] Add a console entry point for mypy

### PR DESCRIPTION
**Pull request by alexandrul**
A possible improvement for #1226

### CODE REVIEW

**Comment by gvanrossum** [comment]
I tried this on OSX using a virtualenv for Python 3.5, and invoking mypy always gives an error:
```
Traceback (most recent call last):
  File "/Users/guido/v3/bin/mypy", line 9, in <module>
    load_entry_point('mypy-lang==0.4.4.dev0', 'console_scripts', 'mypy')()
TypeError: main() missing 1 required positional argument: 'script_path'
```
I don't know the first thing about entry points but I don't think this is an improvement. :-)

**Comment by alexandrul** [comment]
True, sorry for the noise, I'll close the PR if I can't fix it tonight.

**Comment by alexandrul** [comment]
Fixed on my PC, but it's more like a proof of concept for `inspect.stack` :smiley:

I'm using only win7 so I can't check all the corner cases. However, if it works, it may simplify a lot the search for the data folders.

File: mypy/main.py [hunk 1]
```diff
@@ -20,6 +21,16 @@
 PY_EXTENSIONS = tuple(PYTHON_EXTENSIONS)
 
 
+def entry_point() -> None:
+    current_module = inspect.stack()[0][1]
```
Comment (gvanrossum) [inline mypy/main.py]: You should be able to use `__file__` for this.

**Comment by alexandrul** [comment]
Done.

**Comment by gvanrossum** [comment]
Hm... I tried the installation again (using a virtualenv on OS X) and there's still something wrong. The main() code now works (e.g. `mypy -h` works) but as soon as I try to type-check a file I get a crash. Debugging this a little bit it seems it cannot find `__builtins__`. Can you look into this?

**Comment by alexandrul** [comment]
I will look into it.

**Comment by alexandrul** [comment]
On hold until typed-ast will install under Windows.

**Comment by gvanrossum** [comment]
Maybe it's time to reconsider this? At least we're now using setuptools!

**Comment by alexandrul** [comment]
I will give it another shot this week and I will reopen this PR if I'll get something usable.

**Comment by gvanrossum** [comment]
Probably best to open a new PR...

--Guido (mobile)

**Comment by alexandrul** [comment]
I have used `0.470` as a base so I can't create a new PR.

I have moved the `typeshed` and `xml` folders under a new package (`mypy_res`) and switched to `package_data`.
Both Travis and AppVeyor are passing, and `mypy.exe` seems to work just fine on my PC.

Before rebasing, do you have some time to review the changes? https://github.com/python/mypy/compare/b0f1ff9...alexandrul-ci:master

**Pull request closed without merging**